File: server/src/models/puppy.rs
use crate::models::owner::Owner;
use crate::schema::puppies;
use serde_derive::{Deserialize, Serialize};
use typescript_definitions::TypeScriptify;

#[derive(
  Queryable, Identifiable, AsChangeset, Serialize, Deserialize, TypeScriptify, Associations, Clone,
)]
#[belongs_to(Owner)]
#[table_name = "puppies"]
pub struct Puppy {
  pub id: i32,
  pub name: String,
  pub breed: String,
  pub age: i32,
  pub owner_id: Option<i32>,
}

#[derive(Insertable, Deserialize, TypeScriptify)]
#[table_name = "puppies"]
pub struct NewPuppy<'a> {
  pub name: &'a str,
  pub breed: &'a str,
  pub age: i32,
  pub owner_id: Option<i32>,
}

File: dotnet-server/Models/Puppy.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;

namespace dotnet_server.Models
{
  public partial class Puppy
  {
    [JsonProperty("id")]
    public int Id { get; set; } = 0;

    [JsonProperty("name")]
    public string Name { get; set; }

    [JsonProperty("breed")]
    public string Breed { get; set; }

    [JsonProperty("age")]
    public int Age { get; set; }

    [JsonProperty("owner_id")]
    public int? OwnerId { get; set; }

    public virtual Owner Owner { get; set; }
  }
}
